legend-engine-xt-elasticsearch/legend-engine-xt-elasticsearch-V7-pure-metamodel/src/main/resources/core_elasticsearch_seven_metamodel/functions/pure_to_elasticsearch.pure

```diff
@@ -945,19 +1024,51 @@ function meta::external::store::elasticsearch::v7::pureToEs::supportedFilterFunc
+    ,pair(x: Function<Any>[1] | $x == meta::pure::tds::desc_String_1__SortInformation_1_,                          meta::external::store::elasticsearch::v7::pureToEs::reactivateFE_FunctionExpression_1__State_1__Pair_1_)
+    ,pair(x: Function<Any>[1] | $x == meta::pure::tds::agg_String_1__FunctionDefinition_1__FunctionDefinition_1__AggregateValue_1_,                          meta::external::store::elasticsearch::v7::pureToEs::reactivateFE_FunctionExpression_1__State_1__Pair_1_)
+    // catch all - delegate to platform evaluation of expression, and assign to variable.  ES will then use the variable on the plan
+    ,pair(x: Function<Any>[1] | true, meta::external::store::elasticsearch::v7::pureToEs::evaluateAsPureExpressionNode_FunctionExpression_1__State_1__Pair_1_)
   ];
 }
 
-function meta::external::store::elasticsearch::v7::pureToEs::reactivateFE(fe: FunctionExpression[1], req: State[1]): Any[*] 
+function meta::external::store::elasticsearch::v7::pureToEs::reactivateFE(fe: FunctionExpression[1], req: State[1]): Pair<List<Any>, State>[1] 
+{
+  pair($fe->cast(@FunctionExpression)->reactivate($req.sq.inScopeVars)->list(), $req);
+}
+
+function meta::external::store::elasticsearch::v7::pureToEs::evaluateAsPureExpressionNode(fe: FunctionExpression[1], req: State[1]): Pair<List<Any>, State>[1] 
 {
-  $fe->cast(@FunctionExpression)->reactivate($req.sq.inScopeVars);
+  let type = $fe.func->functionReturnType();
+  let multiplicity = $fe.func->functionReturnMultiplicity();
+
+  // todo any validation to prevent compilation errors??
+  // let variables = $fe->findVariableExpressionsInValueSpecification();
+
+  let expression = $fe->printFunctionExpression('');
+  let codeHash    = $expression->hash(HashType.SHA256);
+  let charToIntMap = newMap([
+      pair('0', 0), pair('1', 1), pair('2', 2), pair('3', 3),
+      pair('4', 4), pair('5', 5), pair('6', 6), pair('7', 7),
+      pair('8', 8), pair('9', 9), pair('a', 10), pair('b', 11),
+      pair('c', 12), pair('d', 13), pair('e', 14), pair('f', 15)
+  ]);
+  let prime    = floor(pow(2, 31) - 1);
+  let varName = '@' + $codeHash->toLower()->chunk(1)->fold({c, h | (31 * $h) + $charToIntMap->get($c)->toOne() + 47}, 0)->mod($prime)->toString();  
+
+  let platformNode = processValueSpecification($fe, ^PlatformPlanGenerationState(inScopeVars = $req.sq.inScopeVars, exeCtx = $req.exeCtx), $req.extensions, $req.debug)
+      ->toOne('Expression currently not supported: ' + $expression);
+
+  let allocNode = ^AllocationExecutionNode(resultType = $platformNode.resultType, varName = $varName, executionNodes = $platformNode);
+
+  pair(
+      list(^PlanVarPlaceHolder(type =  $platformNode.resultType.type, name = $varName, multiplicity = $fe.func->functionReturnMultiplicity())),
+      ^$req(extraNodes = $req.extraNodes->concatenate($allocNode))
+  );
```

Comment:
This is the main function to re-write the expression as a `let` and Elasticsearch will just reference the plan variable.